### ISSUE TITLE
Ranks not showing, several errors in log.

### ISSUE THREAD

**Issue by ramadan8** [shavitush/bhoptimer#1161]
## Issue

I'd expect for my rank to show in the scoreboard and in chat, but it appears that I am unranked even when I have records.

![](https://i.imgur.com/2sNWTMo.png)

![](https://i.imgur.com/A9FaxvL.png)

## Logs

```
L 07/27/2022 - 16:37:39: SourceMod error session started
L 07/27/2022 - 16:37:39: Info (map "bhop_aztec") (file "/home/linuxgsm/serverfiles/csgo/addons/sourcemod/logs/errors_20220727.log")
L 07/27/2022 - 16:37:39: [SM] Unable to load plugin "shavit-zones-http.smx": Native "HTTPRequest.HTTPRequest" was not found
L 07/27/2022 - 16:37:52: Error log file session closed.
L 07/27/2022 - 16:37:56: SourceMod error session started
L 07/27/2022 - 16:37:56: Info (map "bhop_aztec") (file "/home/linuxgsm/serverfiles/csgo/addons/sourcemod/logs/errors_20220727.log")
L 07/27/2022 - 16:37:56: [shavit-rankings.smx] Timer (rankings) error 3/4. Reason: You do not have the SUPER privilege and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
```

Before this I also had an issue with the function `GetWeightedPoints` not working, but I fixed it according to [this response](https://github.com/shavitush/bhoptimer/issues/596#issuecomment-385034785) on another issue.

## Game

CS:GO, latest Steam version

## OS

Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye

Although it is running within a Docker container from `gameservermanagers/linuxgsm-docker:latest`.

## Versions

SourceMod: 1.12
MetaMod Source: 1.12

**Comment by Nairdaa** [comment]
SteamWorks seems to be at fault in the first error.

Second one give you the direct answer: "You do not have the SUPER privilege and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)".

On top of that, it seems that you've had issues with your database before and you were able to fix them all by yourself, so I'm assuming you have a complete access to the database. Just grant the SUPER privilege to the timer database user and voila.

**Comment by Nairdaa** [comment]
Also, https://github.com/shavitush/bhoptimer/pull/1162/files I blame 🥑

**Comment by ramadan8** [comment]
> SteamWorks seems to be at fault in the first error.
> 
> Second one give you the direct answer: "You do not have the SUPER privilege and binary logging is enabled (you _might_ want to use the less safe log_bin_trust_function_creators variable)".
> 
> On top of that, it seems that you've had issues with your database before and you were able to fix them all by yourself, so I'm assuming you have a complete access to the database. Just grant the SUPER privilege to the timer database user and voila.
> 
> Also: "SourceMod: 1.12"
> 
> Sourcemod 1.10 is required to run the timer. What says here: https://github.com/shavitush/bhoptimer is fucking misleading and I can't blame you.

Thanks, this fixed it. Although still curious as to why only 1.10 is supported 😄

**Comment by rtldg** [comment]
Some servers run bhoptimer + Sourcemod 1.11 so it *should* work fine afaik, excluding eventqueuefix, which needs to be updated.
Sourcemod 1.10 isn't required, although it is currently the only one I test on.
Sourcemod 1.12 should work fine too since at this point it's only 1.11 with some extra things.

As for why you you were getting:
```
[shavit-rankings.smx] Timer (rankings) error 3/4. Reason: You do not have the SUPER privilege and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
```
I've got no clue why that fixed it for you after downgrading to 1.10.

**Comment by ramadan8** [comment]
> 

I set `log_bin_trust_function_creators` in my MySQL install too ^^
